api/src/main/java/org/openmrs/module/attachments/obs/AbstractAttachmentHandler.java

```diff
@@ -29,10 +31,11 @@
 	public final static String THUMBNAIL_SUFFIX = "_thumb";
 	
 	protected final Log log = LogFactory.getLog(getClass());
-
-	ComplexViewHelper viewHelper = Context.getRegisteredComponent(AttachmentsConstants.COMPONENT_COMPLEXVIEW_HELPER,
-	    ComplexViewHelper.class);
 	
+	@Autowired
+	@Qualifier(AttachmentsConstants.COMPONENT_COMPLEXVIEW_HELPER)
+	private ComplexViewHelper viewHelper;
```

Comment:
@mks-d I am failing to get a hold of the complexViewHelper implementations....I have also tried using the AttachmentContext to provide it and still i bump into an error....what might i be missing?